app/src/main/java/com/concordium/wallet/data/preferences/Preferences.kt

```diff
@@ -15,9 +19,145 @@ open class Preferences {
     interface Listener {
         fun onChange()
     }
+    companion object{
+        const val SHARED_PREFERENCES_ARE_ENCRYPTED ="encrypted_shared_preference"
+    }
 
     constructor(context: Context, preferenceName: String, preferenceMode: Int) {
-        sharedPreferences = context.getSharedPreferences(preferenceName, preferenceMode)
+        val authPreferences = initializeEncryptedSharedPreferences(context, SharedPreferencesKeys.PREF_FILE_AUTH)
+        if(authPreferences.getBoolean(SHARED_PREFERENCES_ARE_ENCRYPTED, false)){
+            sharedPreferences = initializeEncryptedSharedPreferences(context, preferenceName)
+        }else{
+            sharedPreferences =
+                if(migratePreferencesSuccess(context, preferenceName, preferenceMode, authPreferences)){
+                    initializeEncryptedSharedPreferences(context, preferenceName)
+                }else{
+                    getSharedPreferences(context, preferenceName, preferenceMode)
+                }
+        }
+    }
+
+    /**
+     * Returns the required SharedPreferences
+     * @return [SharedPreferences] *if* unencrypted data is still present
+     *
+     * *else* an instance of encrypted [SharedPreferences]
+     */
+    private fun getSharedPreferences(
+        context: Context,
+        preferenceName: String,
+        preferenceMode: Int
+    ): SharedPreferences {
+        val unEncryptedSharedPreferences = context.getSharedPreferences(preferenceName, preferenceMode)
+
+        return if(unEncryptedSharedPreferences.all.isEmpty()) {
+            initializeEncryptedSharedPreferences(context, preferenceName)
+        }else{
+            unEncryptedSharedPreferences
+        }
+    }
+
+    /**
+     * Loops through all of the [SharedPreferences] and attempts to save them to [EncryptedSharedPreferences]
+     * @return *true* if the [SharedPreferences] don't need migrating
+     *
+     * *true* if [SharedPreferences] is successfully migrated and the old data cleared
+     *
+     * *false* if the migrations fail or is not cleared successfully
+     */
+    private fun migratePreferencesSuccess(
```

Comment:
Why is it named "success"? It should go without saying that we're trying to do things successfully, no?

Reply:
It returns Boolean, it tell us if the migration is a success or not.

Reply:
If we look at where the code is called:
` if(migratePreferencesSuccess(context, preferenceName, preferenceMode, authPreferences)){
                initializeEncryptedSharedPreferences(context, preferenceName)
            }else{
                getSharedPreferences(context, preferenceName, preferenceMode)
            }`
we can see that we are only interested in the status of a single SharedPreference with the ` preferenceName`, all of the others and their status are not important, we only need the status of the current one so that we know if we need to use the encrypted or unencrypted SharedPreferences.

Reply:
I will rename it and add to the description.